### ISSUE TITLE
Dependency dashboard is confused about which branches have been created

### ISSUE THREAD

**Issue by wolfgang42** [renovatebot/renovate#6329]
**Describe the bug**
I just set up renovate on a repo, merged the setup MR, and approved one of the updates in the master issue. The next time Renovate ran it updated the master issue description with this:

```
## Open

- [ ] <!-- rebase-branch=renovate/chai-4.x -->Update dependency chai to v4.2.0
- [ ] <!-- rebase-branch=renovate/cheerio-1.x -->Update dependency cheerio to v1.0.0-rc.3
- [ ] <!-- rebase-branch=renovate/mocha-5.x -->Update dependency mocha to v5.2.0
- [ ] <!-- rebase-branch=renovate/docker-ruby-2.x -->[Update ruby Docker tag to v2.7](../merge_requests/28)
- [ ] <!-- rebase-branch=renovate/chai-http-4.x -->Update dependency chai-http to v4
- [ ] <!-- rebase-branch=renovate/mocha-7.x -->Update dependency mocha to v7
- [ ] <!-- rebase-all-open-prs -->**Check this option to rebase all the above open MRs at once**
```
Other than the ruby docker tag (which is the one I approved), none of those other branches exist and they never have. As far as I can tell, Renovate hasn't actually *done* anything with them (which is good, since I didn't ask it to), so it's not clear why they're showing as open instead of pending approval. Weirdly, a bunch of the *other* outdated dependencies are still where they should be, under "Pending Approval".

**What Renovate type are you using?**
 Self-hosted, with GitLab CE.

**Did you see anything helpful in debug logs?**
 Relevant extract of debug log here: https://gist.github.com/wolfgang42/0132e12f63eecd2a3f0bc0ce983a5258

The common thread here seems to be that all of the ones that show up in the wrong place are "blocked by a Pin PR". I'm not entirely clear on what that means, since there are a couple of other dependencies that are planned to be pinned but that's not blocking them? Certainly there are no other MRs that Renovate has created.

**To Reproduce**
This is a rather big and complicated repo. I'm hoping you can work out what happened from the debug log; if not I can have a go at minimizing it but I may not have time to do that for a while.

**Comment by rarkins** [comment]
I have seen this occur before and know why it's happening (although fixing the logic is complicated). So a reproduction is not necessary.

**Comment by HonkingGoose** [comment]
@rarkins maybe change the issue title as we've changed the name of this feature from `master issue` -> `dependency dashboard`?

**Comment by rarkins** [comment]
This should now be fixed in renovate@25.37.1
